lib/datasets.go

```diff
@@ -138,11 +150,19 @@ func (r *DatasetRequests) Save(p *SaveParams, res *repo.DatasetRef) (err error)
 		ds = dsf
 	}
 
+	if ds.Name == "" {
+		return fmt.Errorf("name is required")
+	}
+	if ds.BodyPath == "" && ds.Body == nil && ds.BodyBytes == nil && ds.Structure == nil && ds.Meta == nil && ds.Viz == nil && ds.Transform == nil {
+		return fmt.Errorf("no changes to save")
+	}
+
 	ref, body, err := actions.SaveDataset(r.node, ds, p.DryRun, true)
```

Comment:
Because of the naming, it's sort of unclear that what you are passing is a datasetPod, and a Pod of only the changes that we want to make on the previous dataset.